onnxruntime/contrib_ops/cpu/tokenizer.cc

```diff
@@ -246,40 +77,36 @@ Tokenizer::Tokenizer(const OpKernelInfo& info) : OpKernel(info) {
   status = info.GetAttrs("separators", separators);
   if (!status.IsOK()) {
     status = info.GetAttr("tokenexp", &tokenexp);
-    ORT_ENFORCE(status.IsOK(), "Either one of the separators OR tokenexp attributes required but none is set");
+    ORT_ENFORCE(status.IsOK(), "Either one of the sepexp OR tokenexp attributes required but none is set");
     ORT_ENFORCE(!tokenexp.empty(), "Expecting a non-empty tokenexp");
   } else {
-    ORT_ENFORCE(!separators.empty(), "Expect at least one item within separators");
+    ORT_ENFORCE(!separators.empty(), "separators must not be empty");
+    char_tokenezation_ = (separators.size() == 1 && separators[0].empty());
   }
 
-  char_tokenezation_ = (separators.size() == 1 &&
-                        separators[0].empty());
-
   ORT_ENFORCE(!char_tokenezation_ || mincharnum_ < 2,
               "mincharnum is too big for char level tokenezation");
 
   // Check if we have separators or tokenexp
   if (!char_tokenezation_) {
     if (!separators.empty()) {
-      std::unique_ptr<SearchData> sd(std::make_unique<SearchData>());
-      std::wstring_convert<std::codecvt_utf8<wchar_t>> converter(conv_error, wconv_error);
-      int priority = 0;  // earlier search patterns get priority
+      re2::RE2::Options options;
+      options.set_longest_match(true);
       for (const auto& sep : separators) {
-        ORT_ENFORCE(!sep.empty(), "No empty separators allowed");
-        std::wstring wsep = converter.from_bytes(sep);
-        ORT_ENFORCE(wsep != wconv_error, "Separator strings contains invalid utf8 chars");
-        bool result = sd->tst_.put(wsep.c_str(), wsep.length(), {wsep.length(), priority});
-        ORT_ENFORCE(result, "duplicate separator detected");
-        ++priority;
+        std::unique_ptr<re2::RE2> regex(new re2::RE2(sep, options));
+        if (!regex->ok()) {
+          ORT_THROW("Can not digest sepexp: ", sep, " ", regex->error());
```

Comment:
>sepexp [](start = 36, length = 6)

Consider changing back to a separator

Reply:
sepexp should be changed back to 'separators'?